kiwmi/desktop/xdg_shell.c

```diff
@@ -429,6 +438,10 @@ xdg_shell_new_surface_notify(struct wl_listener *listener, void *data)
     wl_signal_add(
         &xdg_surface->toplevel->events.request_resize, &view->request_resize);
 
+    view->title_change.notify = xdg_toplevel_title_change_notify;
+    wl_signal_add(
+        &xdg_surface->toplevel->events.set_title, &view->title_change);
```

Comment:
The listener & callback are usually named after the event:

```suggestion
    view->set_title = xdg_toplevel_set_title_notify;
    wl_signal_add(
        &xdg_surface->toplevel->events.set_title, &view->set_title);
```

(of course you can then name the new event however you want)

Reply:
Sorry, this comment slipped under my radar.  Changed now.